### PR TITLE
[SPARK-50079][SQL] Assign appropriate error condition for `_LEGACY_ERROR_TEMP_2013`: `NEGATIVE_VALUES_IN_FREQUENCY_EXPRESSION`

### PR DESCRIPTION
**Pull request by itholic**
### What changes were proposed in this pull request?

This PR proposes to assign proper error condition & sqlstate for `_LEGACY_ERROR_TEMP_2013`: `NEGATIVE_VALUES_IN_FREQUENCY_EXPRESSION`

### Why are the changes needed?

To improve the error message by assigning proper error condition and SQLSTATE

### Does this PR introduce _any_ user-facing change?

No, only user-facing error message improved

### How was this patch tested?

Updated the existing tests

### Was this patch authored or co-authored using generative AI tooling?

No

### CODE REVIEW

File: common/utils/src/main/resources/error/error-conditions.json [hunk 1]
```diff
@@ -3521,6 +3521,12 @@
     ],
     "sqlState" : "0A000"
   },
+  "NEGATIVE_VALUES_IN_FREQUENCY_EXPRESSION" : {
+    "message" : [
+      "Negative values found in <frequencyExpression>"
```
Comment (MaxGekk) [inline common/utils/src/main/resources/error/error-conditions.json:3526]: Let's help users in trouble shooting, and show the value:
```suggestion
      "Found the negative value in <frequencyExpression>: <negativeValue>, but expected a positive integral value."
```
just pass the value at:
```scala
      } else if (frqLong < 0) {
        throw QueryExecutionErrors.negativeValueUnexpectedError(frequencyExpression)
      }
```

Comment (itholic) [reply common/utils/src/main/resources/error/error-conditions.json]: Sounds good. Just applied the comments. Thanks!

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/aggregate/percentiles.scala [hunk 2]
```diff
@@ -133,7 +133,8 @@ abstract class PercentileBase
       if (frqLong > 0) {
         buffer.changeValue(key, frqLong, _ + frqLong)
       } else if (frqLong < 0) {
-        throw QueryExecutionErrors.negativeValueUnexpectedError(frequencyExpression)
+        throw QueryExecutionErrors.negativeValueUnexpectedError(
+          frequencyExpression, frqLong.toString)
```
Comment (MaxGekk) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/aggregate/percentiles.scala:137]: Please, just pass a `Long`, and convert it inside of `negativeValueUnexpectedError` properly.

File: sql/catalyst/src/main/scala/org/apache/spark/sql/errors/QueryExecutionErrors.scala [hunk 3]
```diff
@@ -385,10 +385,11 @@ private[sql] object QueryExecutionErrors extends QueryErrorsBase with ExecutionE
...
 
   def negativeValueUnexpectedError(
-      frequencyExpression : Expression): SparkIllegalArgumentException = {
+      frequencyExpression: Expression, negativeValue: String): SparkIllegalArgumentException = {
     new SparkIllegalArgumentException(
-      errorClass = "_LEGACY_ERROR_TEMP_2013",
-      messageParameters = Map("frequencyExpression" -> frequencyExpression.sql))
+      errorClass = "NEGATIVE_VALUES_IN_FREQUENCY_EXPRESSION",
+      messageParameters = Map(
+        "frequencyExpression" -> frequencyExpression.sql, "negativeValue" -> negativeValue))
```
Comment (MaxGekk) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/errors/QueryExecutionErrors.scala:392]: Use `toSQLExpr` for expressions, and `toSQLValue` for the value.

**Review by MaxGekk — CHANGES_REQUESTED** [review]

**Comment by itholic** [comment]
Thanks @MaxGekk for the review. Just applied the comments

**Review by MaxGekk — APPROVED** [review]

**Comment by MaxGekk** [comment]
+1, LGTM. Merging to master.
Thank you, @itholic.

**Pull request closed without merging**